source/language/types.rst

```diff
@@ -114,6 +152,23 @@ conversion will be done assuming little-endian bit ordering.
    // Declare a 32 bit signed integer
    int[32] my_int;
 
+Needs clarification:
+--------------------
+If OpenQASM supports "classical types", then it implicitly declares
+-------------------------------------------------------------------
+support for normative, known fixed-size C types.  Paragraph above seems
+-----------------------------------------------------------------------
+inconsistent.
+-------------
+Can Bit Arrays be cast (implicitly or explicitly) to an Integer type
+--------------------------------------------------------------------
+or a Floating-Point type?
+-------------------------
+What happens during a cast between two scalar types of different bit width?
+---------------------------------------------------------------------------
+What is the bit width of the resulting type?
```

Comment:
This last question seems like it is determined by the cast. i.e. if I write `int[16](x)` where `x` is an `uint[8]` then I am explicitly asking for 16-bit integer output.

Reply:
What about the case where there's loss:

```
uint[16] x;
int[8](x);
```

Possibilities:

- We throw an error because the cast to 8 bits from 16 bits won't fit.
- We silently truncate and continue with potentially catastrophic results.
- We truncate, we issue a warning and we continue with potentially catastrophic results.

If we truncate:

- Which set of bits does the resulting cast get? The most significant bits or the least significant bits?